apps/system/utils/fscmd.c

```diff
@@ -846,7 +852,21 @@ static int tash_mkrd(int argc, char **args)
 
 	/* Then register the ramdisk */
 
-	ret = ramdisk_register(minor, buffer, nsectors, sectsize, RDFLAG_WRENABLED | RDFLAG_FUNLINK);
+	fd = open(FSCMD_DRVPATH, O_RDWR);
+	if (fd < 0) {
+		FSCMD_OUTPUT(CMD_FAILED, "open", FSCMD_DRVPATH);
+	}
+
+	struct ramdisk_info_s info;
+	info.minor = minor;
+	info.buffer = buffer;
+	info.nsectors = nsectors;
+	info.sectsize = sectsize;
+	info.rdflags = RDFLAG_WRENABLED | RDFLAG_FUNLINK;
+
+	ret = ioctl(fd, FCIOC_RAMDISK_REGISTER, (unsigned long)&info);
```

Comment:
Exception handling is required for invalid fd.

Reply:
I will modify

return ERROR